prov/sm2/src/sm2_msg.c

```diff
@@ -319,7 +311,8 @@ static ssize_t sm2_generic_sendmsg(struct sm2_ep *ep, const struct iovec *iov,
 	total_len = ofi_total_iov_len(iov, iov_count);
 	assert(!(op_flags & FI_INJECT) || total_len <= SM2_INJECT_SIZE);
 
-	proto = sm2_select_proto(total_len);
+	proto = sm2_select_proto(desc, iov_count, op_flags, total_len);
+
 	ret = sm2_proto_ops[proto](ep, peer_smr, peer_gid, op, tag, data,
 				   op_flags, mr, iov, iov_count, total_len,
 				   context);
```

Comment:
Line 322... this is as close as it will let me make a comment on...  We need to add sm2_proto_ipc here... Another better alternative solution is to pass op_flags as a pointer, so we don't need to keep extending it... if you choose to do it this way, please make the initial change for the other protocols in a separate commit.
```
if (!(op_flags & FI_DELIVERY_COMPLETE) && proto != sm2_proto_cma) {
```

Reply:
went with the pointer approach and put that in a separate commit